r4/metadata/tests/test_service.bal

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+import health.fhir.templates.r4.metadata.constants;
+import wso2healthcare/healthcare.fhir.r4;
+
+# A resource for generating capability statement
+# + return - capability statement as a json
+function testError() returns r4:FHIRError { 
+    error response = error("Test Error");
+    return r4:createFHIRError(response.message(), r4:FATAL, r4:TRANSIENT_EXCEPTION, response.detail().toString(), cause = response);
+}
+
+@test:Config { groups: ["Service"] }
+function testService() returns error? {
+    http:Client clientEndpoint = check new ("http://localhost:9090");
+    json|error response = clientEndpoint->get("/fhir/r4/metadata");
+    if response is json {
+        test:assertTrue(true);
+    } else {
+        test:assertTrue(false);
+    }
```

Comment:
Since the content of the json is not considered for the test, this can be simplified. If an error occurred, the test will fail.

```suggestion
    _ = check clientEndpoint->get("/fhir/r4/metadata");
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.templates/commit/8dacef8d0685638ff61ce7cb1398abe9236ed812